docs/DeveloperGuide.adoc

```diff
@@ -297,86 +299,263 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+Problem will be deleted and removed from local storage of AlgoBase and cannot find from relevant tags.
 
-_{More to be added}_
+*MSS*
 
-[appendix]
-== Glossary
+1.  User launches AlgoBase.
+2.  User deletes an existing problem by entering index.
+3.  AlgoBase indicates successful deletion of the existing problem.
+4.  AlgoBase updates the UI with the remaining problems.
++
+Use case ends.
 
-[[mainstream-os]] Mainstream OS::
-Windows, Linux, Unix, OS-X
+*Extensions*
+[none]
+* 2a. AlgoBase detects the index is out of bounds.
++
+[none]
+** 2a1. AlgoBase informs user that the deletion is unsuccessful because the index is out of bounds.
++
+Use case ends.
+
+[discrete]
+=== Use Case 4: List Problem
+
+*Guarantees*
+
+A list of existing problems will be returned.
+
+*MSS*
+
+1.  User launches AlgoBase.
+2.  User requests for a list of all existing problems.
+3.  AlgoBase returns a list of problems stored in AlgoBase.
++
+Use case ends.
 
-[[private-contact-detail]] Private contact detail::
-A contact detail that is not meant to be shared with others
+*Extensions*
+[none]
+* 2a. There is no existing problems stored in AlgoBase.
++
+[none]
+** 2a1. AlgoBase informs user that there is no existing problems.
++
+Use case ends.
+
+[discrete]
+=== Use Case 5: Find Problem
+
+*Guarantees*
+
+A list of existing problems with matching fields will be returned.
+
+*MSS*
+
+1.  User launches AlgoBase.
+2.  User requests to find a problem by specifying keywords in certain fields.
+3.  AlgoBase returns a list of problems that have matching keywords in specified fields.
++
+Use case ends.
+
+*Extensions*
+[none]
+* 2a. The user specifies no keywords in any fields.
++
+[none]
+** 2a1. AlgoBase returns a list of all existing problems.
++
+Use case ends.
+
+[none]
+* 2b. There is no problem that matches the user’s requirement.
++
+[none]
+** 2b1. AlgoBase informs user there is no matching problem.
++
+Use case ends.
+
+[none]
+* 2c. There is no existing problems stored in AlgoBase.
++
+[none]
+** 2c1. AlgoBase informs user that there is currently no problems.
++
+Use case ends.
+
+[discrete]
+=== Use Case 6: Sort Problem
+
+*Guarantees*
+
+A list of problems will be returned in a specific order restricted by user.
+
+*MSS*
+
+1.  User launches AlgoBase.
+2.  User requests to sort a set of problems by specifying rules of ordering.
+3.  AlgoBase show the set of questions in sorted order.
++
+Use case ends.
+
+*Extensions*
+[none]
+* 2a. The user specifies no sort order.
++
+[none]
+** 2a1. AlgoBase informs user that no sort order is given.
++
+Use case ends.
+
+[none]
+* 2b. There is no existing problems stored in AlgoBase.
++
+[none]
+** 2b1. AlgoBase informs user that there is currently no problems.
++
+Use case ends.
 
 [appendix]
-== Product Survey
+== Non Functional Requirements
+
+.  Every change is saved immediately and no manual saving is needed.
```

Comment:
You could consider specifying the actual time interval/condition for auto-saving to occur, instead of using "immediately".